sympy/physics/continuum_mechanics/beam.py

```diff
@@ -170,7 +179,21 @@ def second_moment(self):
 
     @second_moment.setter
     def second_moment(self, i):
-        self._second_moment = sympify(i)
+        if isinstance(i, GeometryEntity):
```

Comment:
I think this logic should go in `__init__` rather than `second_moment.setter`.

Reply:
I guess then it would have an impact on the `Beam3D` class, as it is not linked with geometries.
Since the second moment setter is overridden in `Beam3D`, it won't create a problem.